plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -350,21 +357,21 @@ export class PersonState {
         } finally {
             clearTimeout(timeout)
         }
-        return undefined
+        return [undefined, Promise.resolve()]
     }
 
     public async merge(
         otherPersonDistinctId: string,
         mergeIntoDistinctId: string,
         teamId: number,
         timestamp: DateTime
-    ): Promise<InternalPerson | undefined> {
+    ): Promise<[InternalPerson | undefined, Promise<void>]> {
         // No reason to alias person against itself. Done by posthog-node when updating user properties
         if (mergeIntoDistinctId === otherPersonDistinctId) {
-            return undefined
+            return [undefined, Promise.resolve()]
         }
         if (isDistinctIdIllegal(mergeIntoDistinctId)) {
-            await captureIngestionWarning(
+            const kafkaAck = captureIngestionWarning(
```

Comment:
I'd recommend changing `captureIngestionWarning` to pass `waitForAck: false`, but can be done in another PR. I was sure I did that already 😕

Reply:
Yeah, I noticed the `waitForAck: true` late Friday, I'm glad we agree, that's coming right up.